ecs_pod_creator.go

```diff
@@ -164,9 +164,11 @@ func (o *ECSPodCreationOptions) validateContainerDefinitions() error {
 		case NetworkModeHost, NetworkModeAWSVPC:
 			for _, pm := range def.PortMappings {
 				containerPort := utility.FromIntPtr(pm.ContainerPort)
-				hostPort := utility.FromIntPtr(pm.HostPort)
-				catcher.ErrorfWhen(pm.HostPort != nil && hostPort != containerPort,
-					"host port '%d' must be omitted or identical to the container port '%d' when network mode is '%s'", hostPort, containerPort, networkMode)
+				if pm.HostPort != nil {
```

Comment:
I think you can just leave this as it was:
```
catcher.ErrorfWhen(pm.HostPort != nil && utility.FromIntPtr(pm.HostPort) != containerPort...
```
This works cause if `pm.HostPort` is `nil`, the second part of the if statement won't even evaluate since it is automatically false.

Reply:
I changed this because the old code is difficult to read.

Reply:
I thought it was cleaner the other way, but I will leave it up to you!